src/util/ajax.js

```diff
@@ -242,6 +242,12 @@ export const makeRequest = function(requestParameters: RequestParameters, callba
     //   some versions (see https://bugs.webkit.org/show_bug.cgi?id=174980#c2)
     // - Requests for resources with the file:// URI scheme don't work with the Fetch API either. In
     //   this case we unconditionally use XHR on the current thread since referrers don't matter.
+    if(/^custom:/.test(requestParameters.url) && isWorker() && self.worker && self.worker.actor) {
+        return self.worker.actor.send('getResource', requestParameters, callback);
+    }
+    if (/^custom:/.test(requestParameters.url) && !isWorker() && config.LOAD_TILES_FUNCTION != null) {
+        return config.LOAD_TILES_FUNCTION(requestParameters, callback);
+    }
```

Comment:
i think we wouldn't want to do the same regex test twice, e.g. something like this?  This way no function is ever called multiple times (even if `.isWorker()` and `.test()` are optimized internally)

```suggestion
    if (/^custom:/.test(requestParameters.url)) {
        if (isWorker()) {
            if (self.worker && self.worker.actor) {
                return self.worker.actor.send('getResource', requestParameters, callback);
            }
        } else {
            if (config.LOAD_TILES_FUNCTION != null) {
                return config.LOAD_TILES_FUNCTION(requestParameters, callback);
            }
        }
    }
```